packages-content-model/roosterjs-content-model-plugins/lib/edit/deleteSteps/deleteList.ts

```diff
@@ -0,0 +1,24 @@
+import { getClosestAncestorBlockGroupIndex } from 'roosterjs-content-model-core';
+import type { DeleteSelectionStep } from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export const deleteList: DeleteSelectionStep = context => {
+    const { paragraph, marker, path } = context.insertPoint;
+
+    if (context.deleteResult == 'nothingToDelete' || context.deleteResult == 'notDeleted') {
+        const index = getClosestAncestorBlockGroupIndex(path, ['ListItem']);
+        const item = path[index];
+        if (
+            index >= 0 &&
+            paragraph.segments[0] == marker &&
+            item.blockGroupType == 'ListItem' &&
+            (paragraph.segments.length == 1 ||
+                (paragraph.segments.length == 2 && paragraph.segments[1].segmentType == 'Br'))
+        ) {
+            item.levels = [];
+            context.deleteResult = 'singleChar';
```

Comment:
I feel here it would be better to use 'range' so that undo snapshot will be added.